src/overlays/actors/ovl_En_Bdfire/z_en_bdfire.c

```diff
@@ -29,21 +32,200 @@ const ActorInit En_Bdfire_InitVars = {
+    s16 phi_v1_2;
+    Player* player = PLAYER;
+    f32 distToBurn;
+    BossDodongo* bossDodongo;
+    s16 i;
+    s16 phi_v1;
+    s32 temp;
+    
+    bossDodongo = ((BossDodongo*)this->actor.parent);
+    this->unk_158 = bossDodongo->unk_1A2;
+    phi_v1_2 = 0;
+    if (this->actor.params == 0) {
+        Audio_PlaySoundGeneral(NA_SE_EN_DODO_K_FIRE - SFX_FLAG, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+    }
+    Math_SmoothStepToF(&this->actor.scale.x, this->unk_188, 0.3f, 0.5f, 0.0f);
+    Actor_SetScale(&this->actor, this->actor.scale.x);
+    if (this->actor.world.pos.x < -1390.0f) {
+        if (this->actor.velocity.x < -10.0f) {
+            this->actor.world.pos.x = -1390.0f;
+            phi_v1_2 = 1;
+        }
+    }
+    if ((this->actor.world.pos.x > -390.0f) && (this->actor.velocity.x > 10.0f)) {
+        this->actor.world.pos.x = -390.0f;
+        phi_v1_2 = 1;
+    }
+    if ((this->actor.world.pos.z > -2804.0f) && (this->actor.velocity.z > 10.0f)) {
+        this->actor.world.pos.z = -2804.0f;
+        phi_v1_2 = 1;
+    }
+    if ((this->actor.world.pos.z < -3804.0f) && (this->actor.velocity.z < -10.0f)) {
+        this->actor.world.pos.z = -3804.0f;
+        phi_v1_2 = 1;
+    }
+    if (phi_v1_2 != 0) {
+        if (this->unk_158 == 0) {
+            this->actor.world.rot.y += 0x4000;
+        } else {
+            this->actor.world.rot.y -= 0x4000;
+        }
+    }
+    if (this->unk_154 == 0) {
+        temp = 0;
+    } else {
+        this->unk_154--;
+        temp = this->unk_154;
+    }
+    if (temp == 0) {
+        Math_SmoothStepToF(&this->unk_18C, 0.0f, 1.0f, 10.0f, 0.0f);
+        if (this->unk_18C < 10.0f) {
+            Actor_Kill(&this->actor);
+            return;
+        }
+    } else if (!player->isBurning) {
+        distToBurn = (this->actor.scale.x * 130.0f) / 4.2000003f;
+        if (this->actor.xyzDistToPlayerSq < SQ(distToBurn)) {
+            for (i = 0; i < 18; i++) {
+                player->flameTimers[i] = Rand_S16Offset(0, 200);
+            }
+            player->isBurning = true;
+            func_8002F6D4(globalCtx, &this->actor, 20.0f, this->actor.world.rot.y, 0.0f, 8);
+            osSyncPrintf("POWER\n");
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bdfire/EnBdfire_Destroy.s")
+void EnBdfire_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBdfire* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bdfire/func_809BC2A4.s")
+    this->unk_156++;
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(&this->actor);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bdfire/func_809BC598.s")
+void EnBdfire_DrawFire(EnBdfire* this, GlobalContext* globalCtx) {
+    static Gfx* D_809BCB10[] = {
+        0x060264E0, 0x060274E0, 0x060284E0, 0x060294E0, 0x0602A4E0, 0x0602B4E0, 0x0602C4E0, 0x0602D4E0,
+    };
```

Comment:
these are probably textures, so Gfx* isnt right. check in a texture viewer though